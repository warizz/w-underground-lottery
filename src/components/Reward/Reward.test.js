import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Uut from './';

describe('snapshot testing', () => {
  it('should match snapshot when no props', () => {
    const wrapper = shallow(<Uut />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with bet but wrong result', () => {
    const props = {
      bets: [
        {
          number: '111',
          price1: 10,
          price2: 20,
          price3: 30,
        },
      ],
      result: {
        firstThree: '123',
        fourthThree: '101',
        secondThree: '456',
        six: '123456',
        thirdThree: '789',
        two: '69',
      },
    };
    const wrapper = shallow(<Uut {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with matched result', () => {
    const props = {
      bets: [
        {
          number: '123',
          price1: 10,
          price2: 20,
          price3: 30,
        },
      ],
      result: {
        firstThree: '123',
        fourthThree: '101',
        secondThree: '456',
        six: '123456',
        thirdThree: '789',
        two: '69',
      },
    };
    const wrapper = shallow(<Uut {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
