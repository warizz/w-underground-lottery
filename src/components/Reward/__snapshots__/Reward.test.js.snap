// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should match snapshot when no props 1`] = `null`;

exports[`snapshot testing should match snapshot with bet but wrong result 1`] = `
<Segment
  color="grey"
  inverted={true}
  textAlign="center"
>
  You lose
</Segment>
`;

exports[`snapshot testing should match snapshot with matched result 1`] = `
<Segment
  color="green"
  inverted={true}
  textAlign="center"
>
  <div>
    <strong>
      You win!
    </strong>
    <p
      key="ถูก ล่าง [123] 30 x 100 = 3000 บาท"
    >
      ถูก ล่าง [123] 30 x 100 = 3000 บาท
    </p>
  </div>
</Segment>
`;
